Sming/Components/rboot/component.mk

```diff
@@ -125,20 +151,50 @@ endif # RBOOT_EMULATION
 
 # Define our flash chunks
 FLASH_RBOOT_BOOT_CHUNKS				:= 0x00000=$(RBOOT_BIN)
-FLASH_RBOOT_APP_CHUNKS				:= $(ROM_0_ADDR)=$(RBOOT_ROM_0_BIN)
+FLASH_RBOOT_APP_CHUNKS				:= $(RBOOT_ROM0_ADDR)=$(RBOOT_ROM_0_BIN)
 FLASH_RBOOT_ERASE_CONFIG_CHUNKS		:= 0x01000=$(SDK_BASE)/bin/blank.bin
 
 ifndef RBOOT_EMULATION
+
+# => Automatic linker script generation from template
+# $1 -> application target
+# $2 -> linker script
+# $3 -> ROM address variable (not value!)
+define GenerateLinkerScriptTargets
+# Mark linker script out-of-date if ROM address differs from previous run
+-include $2.config
+ifneq ($$(GEN_$3),$$($3))
+.PHONY: $2
+endif
+# Generate linker script from template
+$2: $(RBOOT_LD_TEMPLATE)
+	$$(info LDGEN $$@)
+	$$(Q) awk 'match($$$$0, /(^.*irom0_0_seg\s*:\s*)\S+/, m) { \
+			printf "%sorg = 0x%08X, len = 1M - 0x%X\n", m[1], 0x40200010 + $$($3) % 0x100000, $$($3) % 0x100000; next \
+		} 1 { print $$$$0 }' $$< > $$@
```

Comment:
mawk doesn't like this, as far as I can tell because it doesn't support the 3rd 'array' parameter.